src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1161,6 +1161,11 @@ private void findSig(CORINFO_MODULE_STRUCT_* module, uint sigTOK, CORINFO_CONTEX
             var methodSig = (MethodSignature)methodIL.GetObject((int)sigTOK);
             Get_CORINFO_SIG_INFO(methodSig, sig);
 
+            if (sig->callConv == CorInfoCallConv.CORINFO_CALLCONV_UNMANAGED)
+            {
+                throw new NotSupportedException();
```

Comment:
Do we know what will be required to ensure it is supported?

Reply:
Likewise, do we need issues for CoreRT/Mono who may want to support this as well?

Reply:
@lambdageek Has been tagged on this above. This specific PR is for adding the value so it can be consumed from the Roslyn side. When it is actually implemented CrossGen2 should be okay to support this if `modopts` are automatically copied over.

Reply:
Do ildasm/ilasm need changes to support the new calling convention?

Reply:
That will also need support for the IL scenario.

Reply:
@BruceForstall or @briansull What is needed to add support in the IL tooling for this new MD value?

Reply:
ilasm supports arbitrary calling convention values. E.g. you can write `.method private hidebysig static callconv(9) void  foo() cil managed` today and ilasm will do the right thing for it.

ildasm is not able to disassembly it though. It errors out with `signature of method '%s' has invalid calling convention`. It would be pretty straightforward to fix that.

It would be nice to teach ilasm/ildasm to print symbolic name for this. That means adding a new line to callKind in `src\coreclr\src\ilasm\asmparse.y` and regenerating asmparse.c using internal version of yacc (it somewhere in the in the  desktop tree - related: https://github.com/dotnet/runtime/issues/4776).

Reply:
@jkotas and @tannergooding I regenerated the updated parser and added the textual grammar. I just made sure it built. I will do some testing locally to validate it works as expected.

/cc @BruceForstall @briansull